banpei/hotelling.py

```diff
@@ -22,16 +22,15 @@ def detect(self, data, threshold):
         data = self.convert_to_nparray(data)
 
         # Set the threshold of abnormality
-        abn_th = stats.chi2.interval(1-threshold, 1)[1]
+        abn_th = stats.chi2.interval(1 - threshold, 1)[1]
 
         # Covert raw data into the degree of abnormality
         avg = np.average(data)
         var = np.var(data)
         data_abn = [(x - avg)**2 / var for x in data]
 
         # Abnormality determination
-        result = []
-        for (index, x) in enumerate(data_abn):
-            if x > abn_th:
-                result.append((index, data[index]))
+        result = np.full_like(data, np.nan)
+        tf_ind = data_abn > abn_th
+        result[tf_ind] = data[tf_ind]
```

Comment:
Originally, since it was implemented so as to return an array of the same size as the input data, a CI error has appeared.  
I am in favor of the change to speed up with ndarray!

Reply:
Yes, I made a bit more modification on my version.

Reply:
Changing the size of the return value means that a program that is working normally will not work by merging your change.    
You should adjust and modify them so that CI tests pass.